modules/azure_aks/variables.tf

```diff
@@ -1,81 +1,123 @@
-variable aks_cluster_name {}
+variable "aks_cluster_name" {
+  description = "The name for the AKS resources created in the specified Azure Resource Group"
+  type        = string
+}
+
+variable "aks_cluster_rg" {
+  description = "The resource group name to be imported"
+  type        = string
+}
 
-variable aks_cluster_rg {}
-variable aks_cluster_rg_id {}
-variable aks_cluster_dns_prefix {}
+variable "aks_cluster_rg_id" {
+  description = "The `azurerm_kubernetes_cluster`'s id."
+  type        = string
+}
+
+variable "aks_cluster_dns_prefix" {
+  description = "DNS prefix specified when creating the managed cluster."
+  type        = string
+}
 
 variable "aks_cluster_location" {
   description = "The Azure Region in which all resources in this example should be provisioned"
+  type        = string
   default     = "eastus"
 }
 
-variable aks_cluster_sku_tier {
+variable "aks_cluster_sku_tier" {
   description = "The SKU Tier that should be used for this Kubernetes Cluster. Possible values are Free and Paid (which includes the Uptime SLA). Defaults to Free"
+  type        = string
   default     = "Free"
 
   validation {
-    condition     = contains(["Free", "Paid"],  var.aks_cluster_sku_tier)
+    condition     = contains(["Free", "Paid"], var.aks_cluster_sku_tier)
     error_message = "ERROR: Valid types are \"Free\" and \"Paid\"!"
   }
 }
 
 variable "aks_private_cluster" {
-  default = false
+  description = "The cluster API endpoint uses Private IP address?"
+  type        = bool
+  default     = false
 }
 
 variable "aks_cluster_node_count" {
-  default = 4
+  description = "(Required, when default_nodepool_auto_scaling=true) The minimum number of nodes which should exist in this Node Pool. If specified this must be between 1 and 100."
+  type        = number
+  default     = 4
 }
 
-variable "aks_availability_zones" {}
+variable "aks_availability_zones" {
+  description = "A list of Availability Zones across which the Node Pool should be spread. Changing this forces a new resource to be created."
+  type        = list(string)
+  default     = ["1"]
+}
 
 # https://docs.microsoft.com/en-us/azure/virtual-machines/linux/sizes
 variable "aks_cluster_node_vm_size" {
-  default = "Standard_D4_v2"
+  description = "The default virtual machine size for the Kubernetes agents"
+  type        = string
+  default     = "Standard_D4_v2"
 }
 
 variable "aks_cluster_node_admin" {
-  default = "ubuntu"
+  description = "The operating system Admin User for VMs of AKS cluster nodes"
```

Comment:
suggestion: use AKS cluster nodes in both descriptions, Kubernetes agents seems like a different object but looks they both refer to AKS cluster nodes.

Reply:
Updated the description in the latest commit.